engines/stark/resources/location.cpp

```diff
@@ -65,6 +67,20 @@ void Location::onAllLoaded() {
 void Location::onGameLoop() {
 	Object::onGameLoop();
 
+	if (_fadeFramesRemaining > 0) {
+		_fadeFramesRemaining--;
+
+		float newFadeLevel = StarkScene->getFadeLevel() + _fadeLevelIncrement;
+		if (newFadeLevel < 1.0) {
```

Comment:
This can be changed to `newFadeLevel = CLIP(StarkScene->getFadeLevel() + _fadeLevelIncrement, 0.0, 1.0);`.